hal/src/nRF52840/i2c_hal.cpp

```diff
@@ -54,37 +59,80 @@ typedef struct {
 
     uint8_t                     address;    // 7bit data
 
-    uint8_t                     rx_buf[BUFFER_LENGTH];
-    uint8_t                     rx_buf_index;
-    uint8_t                     rx_buf_length;
-    uint8_t                     tx_buf[BUFFER_LENGTH];
-    uint8_t                     tx_buf_index;
-    uint8_t                     tx_buf_length;
+    volatile uint8_t            rx_buf[BUFFER_LENGTH];
```

Comment:
I don't see any reason to make all of these volatile. In slave mode, they are going to be accessed only from interrupt context in `onRequest()` and `onReceive()` callbacks.

Reply:
Removed `volatile` for `rx_buf_index` and `rx_buf_length`